torch/cuda/__init__.py

```diff
@@ -64,6 +64,26 @@ def is_available():
     if (not hasattr(torch._C, '_cuda_isDriverSufficient') or
             not torch._C._cuda_isDriverSufficient()):
         return False
+    # Lazy initialization here is necessary, as once we
+    # call cudaGetDeviceCount() we have initialized the CUDA
+    # API and forks are no longer permissible.  Calling
+    # _lazy_init() is required so we can give a good error
+    # message in this case.
+    #
+    # However, the way we have written the initialization here
+    # is a bit bogus, as the very act of checking if
+    # the driver is sufficient will initialize the CUDA
```

Comment:
no, the act of checking if the driver is sufficient does not initialize the CUDA runtime.
The act of querying the device count initializes the driver.

previously, we used to return the value of a driver check, but not check if `deviceCount > 0`.
Arguably that's a better default than `getDeviceCount > 0` if the result of `getDeviceCount > 0` results in the cudacontext being initialized...

I'm a bit conflicted.

Reply:
For reference, we changed this behavior in 767c96850da0d55a567e91368971ca74a1d302f0